webodf/tools/odfRng2Config.js

```diff
@@ -22,28 +22,73 @@
  * @source: https://github.com/kogmbh/WebODF/
  */
 
-function aggregate(callback) {
-    return function (collection, individual) {
-        return collection.concat(callback(individual));
-    };
-}
+"use strict";
 
-function toArray(nodeList) {
-    "use strict";
-    return Array.prototype.slice.call(nodeList);
+var rngns = "http://relaxng.org/ns/structure/1.0";
+
+/**
+ * Check if a node is an element in the rng namespace and the given localName.
+ * @param {!string} localName
+ * @param {!Node} node
+ * @return {!boolean}
+ */
+function isRng(localName, node) {
+    if (node.nodeType !== 1) {
+        return false;
+    }
+    if (node.namespaceURI !== rngns) {
+        return false;
+    }
+    if (node.localName !== localName) {
+        return false;
+    }
+    return true;
 }
 
-function getName(node) {
-    return node && node.getAttribute("name");
+/**
+ * @param {!Node} node
+ * @return {?Element}
+ */
+function getFirstElementNode(node) {
+    node = node.firstChild;
```

Comment:
There's a ``firstElementChild`` and ``nextElementSibling`` properties you might be able to use here instead?